src/gpu/SDL_gpu.c

```diff
@@ -2796,3 +2796,16 @@ void SDL_ReleaseGPUFence(
         device->driverData,
         fence);
 }
+
+Uint32 SDL_CalculateGPUTextureFormatSize(
+    SDL_GPUTextureFormat format,
+    Uint32 w,
+    Uint32 h,
+    Uint32 d)
+{
+    Uint32 pixelRowsPerBlock = Texture_GetBlockSize(format);
+    Uint32 pixelColumnsPerBlock = pixelRowsPerBlock;
```

Comment:
Is this accurate for non-square block sizes?

Reply:
You're right, we need to restructure this for ASTC.